File: age_calculator.pl
#!/usr/bin/perl
# Perl Age Calculator
# Run with: perl age_calculator.pl

use strict;
use warnings;
use POSIX qw(strftime);

sub get_age_message {
    my ($age) = @_;

    if ($age < 30) {
        return "You are young!";
    } elsif ($age >= 30 && $age < 67) {
        return "You are middle-aged.";
    } elsif ($age >= 67 && $age <= 80) {
        return "You are aging and should travel the world!";
    } else {
        return "You should relax and enjoy your life!";
    }
}

my $run_again = "yes";

while ($run_again eq "yes" || $run_again eq "y") {
    # Get current year
    my $current_year = strftime("%Y", localtime);

    # Welcome message
    print "\n========================================\n";
    print "   Welcome to the Age Calculator!\n";
    print "========================================\n\n";

    # Get user input
    print "Please enter your name: ";
    chomp(my $name = <STDIN>);

    print "Please enter your year of birth: ";
    chomp(my $birth_year = <STDIN>);

    # Calculate age
    my $age = $current_year - $birth_year;

    # Get appropriate message
    my $message = get_age_message($age);

    # Output results
    print "\n----------------------------------------\n";
    print "Hello, $name!\n";
    print "You are $age years old.\n";
    print "$message\n";
    print "----------------------------------------\n\n";

    # Ask to run again
    print "Would you like to run the program again? (yes/no): ";
    chomp($run_again = <STDIN>);
    $run_again = lc($run_again);
}

print "\nThank you for using the Age Calculator. Goodbye!\n";

File: age_calculator.py
# Python Age Calculator
# Run with: python age_calculator.py

from datetime import datetime

def get_age_message(age):
    if age < 30:
        return "You are young!"
    elif 30 <= age < 67:
        return "You are middle-aged."
    elif 67 <= age <= 80:
        return "You are aging and should travel the world!"
    else:
        return "You should relax and enjoy your life!"

def main():
    while True:
        # Get current year
        current_year = datetime.now().year

        # Welcome message
        print("\n========================================")
        print("   Welcome to the Age Calculator!")
        print("========================================\n")

        # Get user input
        name = input("Please enter your name: ")
        birth_year = int(input("Please enter your year of birth: "))

        # Calculate age
        age = current_year - birth_year

        # Get appropriate message
        message = get_age_message(age)

        # Output results
        print("\n----------------------------------------")
        print(f"Hello, {name}!")
        print(f"You are {age} years old.")
        print(message)
        print("----------------------------------------\n")

        # Ask to run again
        run_again = input("Would you like to run the program again? (yes/no): ").lower()
        if run_again not in ["yes", "y"]:
            break

    print("\nThank you for using the Age Calculator. Goodbye!")

if __name__ == "__main__":
    main()
